code/modules/vending/_vending.dm

```diff
@@ -759,33 +790,200 @@ GLOBAL_LIST_EMPTY(vending_products)
+
+			if(!C)
+				say("No card found.")
+				flick(icon_deny,src)
+				vend_ready = 1
+				return
+			else if (!C.registered_account)
+				say("No account found.")
+				flick(icon_deny,src)
+				vend_ready = 1
+				return
+			var/datum/bank_account/account = C.registered_account
+			for(var/obj/O in contents)
+				if(O.name == N)
+					S = O
+					break
+			if(!account.has_money(S.custom_price))
+				say("You do not possess the funds to purchase this.")
+			else
+				account.adjust_money(-S.custom_price)
+				var/datum/bank_account/owner = private_a
+				if(owner)
+					owner.adjust_money(S.custom_price)
+				use_power(5)
+				vending_machine_input[N] = max(vending_machine_input[N] - 1, 0)
+				if(last_shopper != usr || purchase_message_cooldown < world.time)
+					say("Thank you for buying local and purchasing [S]!")
+					purchase_message_cooldown = world.time + 5 SECONDS
+					last_shopper = usr
+				S.forceMove(drop_location())
+				loaded_items--
+				vend_ready = 1
+	updateUsrDialog()
+
+/obj/machinery/vending/custom/ui_interact(mob/user)
+	var/list/dat = list()
+	var/datum/bank_account/account
+	var/mob/living/carbon/human/H
+	var/obj/item/card/id/C
+	if(ishuman(user))
+		H = user
+		C = H.get_idcard(TRUE)
+	///temp var to set the shown price
+	var/price = 1
+
+	if(!C)
+		dat += "<font color = 'red'><h3>No ID Card detected!</h3></font>"
+	else if (!C.registered_account)
+		dat += "<font color = 'red'><h3>No account on registered ID card!</h3></font>"
+	else
+		account = C.registered_account
+	if(vending_machine_input.len)
+		dat += "<h3>[input_display_header]</h3>"
+		dat += "<div class='statusDisplay'>"
+		for (var/O in vending_machine_input)
+			if(vending_machine_input[O] > 0)
+				var/N = vending_machine_input[O]
+				dat += "<a href='byond://?src=[REF(src)];dispense=[sanitize(O)]'>Dispense</A> "
+				if(compartmentLoadAccessCheck(user))
+					price = "FREE"
+				else
+					for(var/obj/T in contents)
+						if(T.name == O)
+							price = "$[T.custom_price]"
+							break
+				dat += "<B>[O] ([price]): [N]</B><br>"
+		dat += "</div>"
+		dat += "<b>Balance: $[account.account_balance]</b>"
+
+	var/datum/browser/popup = new(user, "vending", (name))
+	popup.add_stylesheet(get_asset_datum(/datum/asset/spritesheet/vending))
+	popup.set_content(dat.Join(""))
+	popup.set_title_image(user.browse_rsc_icon(icon, icon_state))
+	popup.open()
+
+/obj/machinery/vending/custom/attackby(obj/item/I, mob/user, params)
+
+	if(!private_a)
+		var/mob/living/carbon/human/H
+		var/obj/item/card/id/C
+		if(ishuman(user))
+			H = user
+			C = H.get_idcard(TRUE)
+			if(C?.registered_account)
+				private_a = C.registered_account
+				say("\The [src] has been linked to [C].")
+
+	if(compartmentLoadAccessCheck(user))
+		if(istype(I, /obj/item/pen))
+			name = stripped_input(user,"Set name","Name", name, 20)
+			desc = stripped_input(user,"Set description","Description", desc, 60)
+			slogan_list += stripped_input(user,"Set slogan","Slogan","Epic", 60)
```

Comment:
can you set the default slogan to "We need a slogan!" for a sick ref to a tarn adams game no one played